doc/modules/tree.rst

```diff
@@ -500,6 +500,37 @@ Entropy:
 
     H(Q_m) = - \sum_k p_{mk} \log(p_{mk})
 
+
+.. note::
+
+  The entropy criterion computes the Shannon entropy of the possible classes. It
+  takes the class frequencies of the training data points that reached a given
+  leaf :math:`m` as their probability. Using the **Shannon entropy as tree node
+  splitting criterion is equivalent to minimizing the log loss** (also known as
+  cross-entropy and multinomial deviance) between the true labels :math:`y_i`
+  and the probalistic predictions :math:`T(x_i)` of the tree model :math:`T`.
+
+  To see, this, first recall that the log loss of a tree model :math:`T`
+  computed on a dataset :math:`D` is defined as follows:
+
+  .. math::
+  
+      \mathrm{LL}(D, T) = -\frac{1}{n} \sum_{(x_i, y_i) \in D} \sum_k I(y_i = k) \log(T(x_i))
+
+  where :math:`D` is a training dataset of :math:`n` pairs :math:`(x_i, y_i)`.
+
+  In a classification tree, the predicted class probabilities within leaf nodes
+  are constant, that is: for all :math:`(x_i, y_i) \in Q_m` such that :math:`y_i
+  = k` then :math:`T(x_i) = p_{mk}`.
```

Comment:
```suggestion
  are constant, that is: for all :math:`(x_i, y_i) \in Q_m` one has
  :math:`T(x_i) = p_{mk}`.
```
`y` is not constant within a leaf node, just the prediction is constant.

Reply:
My notation is indeed bad. I would probably index `T(x_i)` by k: `T(x_i)_k` to denote the k-th element of the vector of class-wise probabilities `T(x_i)` predicted for by the tree `T` for input features `x_i`.

Reply:
done in 32542f4. @lorentzenchr feel free to merge if this is good for you.